### PR TITLE
Refactors existing modal and adds new error modal

### PR DESCRIPTION
**Pull request by sweir27**
This PR:
- Extracts the "meaty" bits of our existing `Modal` component into a `ModalWrapper` so that it can be re-used for a `ModalError` (named to keep it consistent, but open to change).
- Updates `Modal` to consume the new `ModalWrapper`
- Adds a new `ModalError` which looks like:
image

I spent a while faffing around trying to extend the existing `Modal` component for this error case, but there was a lot that was coupled in there. The error modal is much simpler and has different styling rules, but it still needs to handle the animations, blur behavior, etc.

If people find that this abstraction doesn't make sense, I am happy to update!

Also... if anyone knows why the `color` isn't being picked up on my `Sans` elements, please let me know. It's coming through in the css as:
image
which suggests that it's not being interpreted correctly but I'm not sure what I'm doing wrong.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 9f3bc402a3fbc66a9ad29b60e521429b9d93394a

https://deploy-preview-1219--artsy-reaction.netlify.com

**Comment by mzikherman** [comment]
I too have gotten those colors making their way in on `Sans` elements! I just hardcoded those for now- https://github.com/artsy/reaction/blob/a3ba885c185bf21c0417d8fbb1298d2f0ee3c80b/src/Apps/WorksForYou/WorksForYouContents.tsx#L136

File: src/Components/Modal/ModalError.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export class ModalError extends React.Component<ModalErrorProps> {
+  static defaultProps = {
+    headerText: "An error occurred",
+    closeText: "Continue",
+  }
+
+  close = () => {
+    this.props.onClose()
```
Comment (mzikherman) [inline src/Components/Modal/ModalError.tsx]: looks like `onClose` is optional, so may need a check before invoking

**Comment by yuki24** [comment]
In order to use keys like `black60` you will have to wrap `<Sans>`s (or any palette component) with a theme provider. Here is [a living example in Emission that sets it up](https://github.com/artsy/emission/blob/master/src/lib/Components/Bidding/Components/BiddingThemeProvider.tsx#L7). I'm not sure where the provider should live in Force, but I'd be happy to pair on that if you want to.

**Comment by zephraph** [comment]
If the theme provider wasn't set up correctly, I wouldn't expect the fonts and other stuff to pull in either... Seems strange to me... Would be good to double check though. @damassi have you seen anything like this?

**Comment by damassi** [comment]
@zephraph - going to move your question to FE

**Comment by zephraph** [comment]
So, after quite a bit of investigation, I think that I can confirm that the color not working **is** due to the context provider not being present or configured correctly. Here's a barebones test case that shows it working fine: https://codesandbox.io/s/7kvk2nov20.

File: src/Components/Modal/ModalWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    const { isShown, isAnimating } = this.state
+
+    if (isShown) {
+      this.addBlurToContainers()
+    } else {
+      this.removeBlurToContainers()
+    }
+
+    return (
+      <Wrapper isShown={isShown || isAnimating}>
```
Comment (zephraph) [inline src/Components/Modal/ModalWrapper.tsx]: I'm assuming that the Modal maybe be rendered in a portal which might be the reason why it's not being recognized as being rendered in the theme provider. To combat that, wrap this in the `Theme` component from palette. The color should work fine after that.

File: src/Components/Modal/ModalError.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+interface ModalErrorProps extends React.HTMLProps<HTMLDivElement> {
+  show?: boolean
+  headerText?: string
+  detailText?: string
+  closeText?: string
+  onClose?: () => void
+}
+
+export class ModalError extends React.Component<ModalErrorProps> {
```
Comment (yuki24) [inline src/Components/Modal/ModalError.tsx]: I wonder if this should be called `ErrorModal`? `ModalError` implies that it's a class that extends `Error` (the error class).

Comment (sweir27) [reply src/Components/Modal/ModalError.tsx]: I was trying to keep everything under `Modal` namespaced that way (`Modal...`). But `ErrorModal` does sound better. I'll update and we can always change.

**Comment by sweir27** [comment]
Thanks for the comments, all. I believe I've addressed them and this is ready for another look!

**Pull request merged**